manifests-gen/providers.go

```diff
@@ -141,6 +182,53 @@ func (p *provider) writeProviderComponentsConfigmap(fileName string, objs []unst
 	return os.WriteFile(path.Join(*manifestsPath, fileName), ensureNewLine(cmYaml), 0600)
 }
 
+// addComponentsToConfigMap adds the components to configMap. The components are compressed if their size would be over 950KB.
+func addComponentsToConfigMap(cm corev1.ConfigMap, objs []unstructured.Unstructured) (corev1.ConfigMap, error) {
+	combined, err := utilyaml.FromUnstructured(objs)
+	if err != nil {
+		return corev1.ConfigMap{}, fmt.Errorf("error converting unstructured object to YAML: %w", err)
+	}
+
+	compressionRequired := len(combined) > 950_000 // 98KB under the configMap 1MiB limit to leave space for rest of the configMap
+	if compressionRequired {
+		compressed, err := compressToZstd(combined)
+		if err != nil {
+			return corev1.ConfigMap{}, fmt.Errorf("failed to compress components: %w", err)
+		}
+
+		cm.BinaryData = map[string][]byte{
+			"components-zstd": compressed.Bytes(),
+		}
+	} else {
+		cm.Data = map[string]string{"components": string(combined)}
+	}
+
+	return cm, nil
+}
+
+// compressToZstd compressses bytes using zstd
+func compressToZstd(data []byte) (bytes.Buffer, error) {
```

Comment:
Is this an exact copy of the (p *provider) compressToZstd above?

Reply:
Removed the other one. I think it accidentally came back during rebase.

Reply:
TY